initial commit and push

merrickfox opened the pull request:
@kulor @adamdunkley here is the code for the new open source email auto-complete, you can preview the functionality here: http://bw-feature-email-auto-fill.s3-website-eu-west-1.amazonaws.com/send-flowers in the email field.

Documentation to follow, I'll write up a nice readme.
The basic outline is you add the ```email-typeahead``` attribute directive to the input box of your choice, it will create a new DOM input element as a sibling and you use a style (position: absolute; top: 0) to overlay it (this is how everyone else seems to implement this functionality on web), styling will be down to the user and they will also have to specify correct z-indices too. From there the directive is pulling what is already typed, if there is an @ symbol it starts to compare the domain against the list of domains and returns the first one that matches for the first characters typed, it then just displays that in the newly created typeahead input element. There are event listeners for ```tab``` and ```enter``` key, if there is a valid suggestion and either of those keys are pressed it will replace the ng-model of the email input element with the email + the suggestion.

kulor commented on LICENSE:
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
+
+Copyright (c) 2015 Merrick Fox
```
The copyright belongs to B&W as you did it on work time for a work project.

merrickfox replied:
Haha sorry, it was auto generated!
On 14 Sep 2015 18:36, "James Broad" <notifications@github.com> wrote:

> In LICENSE
> <https://github.com/BloomAndWild/angular-bw-email-autofill/pull/1#discussion_r39422179>
> :
>
> > @@ -0,0 +1,22 @@
> > +The MIT License (MIT)
> > +
> > +Copyright (c) 2015 Merrick Fox
>
> The copyright belongs to B&W as you did it on work time for a work project.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BloomAndWild/angular-bw-email-autofill/pull/1/files#r39422179>
> .
>

kulor replied:
Sorry, that probably came across arsey, not intended.

kulor commented on bower.json:
```diff
@@ -0,0 +1,17 @@
+    "bower_components",
+    "test",
+    "tests"
+  ]
```
this package has a dependency on lodash and angular, they should be added to the dependency list

kulor commented:
It'd be great to see an isolated working demo page for this, either in the repo or as a linked JSFiddle/JSBin/Codepen.

merrickfox commented:
Addressed comments, started the README, I'm going to merge this to master so I can register the module with Bower, then continue making adjustments and add documentation,

kulor commented:
@merrickfox while this PR is closed, I still feel it important to have a working example of this code somewhere, It'll reduce the barrier to entry, on setup configuration if nothing else.
